src/models/Settings.php

```diff
@@ -10,29 +10,77 @@
 use craft\base\Model;
 use craft\behaviors\EnvAttributeParserBehavior;
 use yii\behaviors\AttributeTypecastBehavior;
+use craft\helpers\UrlHelper;
 /**
  * @author    Website Toolbox
  * @package   Websitetoolboxforum
- * @since     3.0.0
+ * @since     4.0.0
  */
 class Settings extends Model
 {
-    // Public Properties
-    // =========================================================================
+    // Public Properties    
     public $forumUsername         = '';
     public $forumPassword         = '';
     public $forumEmbedded         = 1;
     public $forumApiKey           = '';
     public $forumUrl              = '';  
+    public $communityUrl          = 'forum';  
   
     // Public Methods
+    protected function makeUrl($page){
+        $siteUrl = UrlHelper::siteUrl();                
+        if(strpos($siteUrl, 'index.php') >= 0){
+            $embedPage = $siteUrl.'?p='.$page;
```

Comment:
Please add comment why are you using ?p= hard coded.

Reply:
The hard codded was added for development purpose. I can be replace by craft dynamic value. I will replace it.